### PR TITLE
luarocks: append RT_DIR arg in wrapper

### LINKED ISSUE #229291: luarocks test -- packaging issue
### Describe the bug
Running `luarocks test` with either `--local` or `--tree /some/path` results in luarocks trying to locally install `luasystem`, which is a dependency of `busted`.

This fails with one two different messages:

```
Error: Failed installing dependency: https://luarocks.org/busted-2.1.2-3.src.rock - Failed installing dependency: https://luarocks.org/luasystem-0.2.1-0.src.rock - Could not find library file for RT
  No file librt.a in /usr/local/lib
  No file librt.a in /usr/local/lib64
  No file librt.so in /usr/local/lib
  No file librt.so in /usr/local/lib64
  No file matching librt.so.* in /usr/local/lib
  No file matching librt.so.* in /usr/local/lib64
  No file librt.a in /usr/lib
  No file librt.a in /usr/lib64
  No file librt.so in /usr/lib
  No file librt.so in /usr/lib64
  No file matching librt.so.* in /usr/lib
  No file matching librt.so.* in /usr/lib64
  No file librt.a in /lib
  No file librt.a in /lib64
  No file librt.so in /lib
  No file librt.so in /lib64
  No file matching librt.so.* in /lib
  No file matching librt.so.* in /lib64
You may have to install RT in your system and/or pass RT_DIR or RT_LIBDIR to the luarocks command.
Example: luarocks install luasystem RT_DIR=/usr/local
```

or

```
Error: LuaRocks 3.9.1 bug (please report at https://github.com/luarocks/luarocks/issues).
Arch.: linux-x86_64
...nyv78liky-luarocks-3.9.1/share/lua/5.2/luarocks/deps.lua:243: assertion failed!
stack traceback:
	[C]: in function 'assert'
	...nyv78liky-luarocks-3.9.1/share/lua/5.2/luarocks/deps.lua:243: in function 'fulfill_dependency'
	...nyv78liky-luarocks-3.9.1/share/lua/5.2/luarocks/deps.lua:336: in function 'fulfill_dependencies'
	...yv78liky-luarocks-3.9.1/share/lua/5.2/luarocks/build.lua:138: in function 'process_dependencies'
	...yv78liky-luarocks-3.9.1/share/lua/5.2/luarocks/build.lua:379: in function 'build_rockspec'
	...liky-luarocks-3.9.1/share/lua/5.2/luarocks/cmd/build.lua:74: in function <...liky-luarocks-3.9.1/share/lua/5.2/luarocks/cmd/build.lua:51>
	(...tail calls...)
	...liky-luarocks-3.9.1/share/lua/5.2/luarocks/cmd/build.lua:154: in function <...liky-luarocks-3.9.1/share/lua/5.2/luarocks/cmd/build.lua:117>
	(...tail calls...)
	...nyv78liky-luarocks-3.9.1/share/lua/5.2/luarocks/deps.lua:237: in function 'fulfill_dependency'
	...ky-luarocks-3.9.1/share/lua/5.2/luarocks/test/busted.lua:24: in function <...ky-luarocks-3.9.1/share/lua/5.2/luarocks/test/busted.lua:19>
	(...tail calls...)
	[C]: in function 'xpcall'
	...mnyv78liky-luarocks-3.9.1/share/lua/5.2/luarocks/cmd.lua:672: in function 'run_command'
	...4hmwbjqr4mnyv78liky-luarocks-3.9.1/bin/.luarocks-wrapped:38: in main chunk
	[C]: in ?
```

### Steps To Reproduce
1. Check out the `luarocks-test` branch of https://github.com/nvim-neorocks/luarocks-tag-release
2. Enter a nix shell with `lua5_1` and `luarocks`
3. In the checked out repository, run `luarocks test --local` or `luarocks test --tree /tmp/testdir` (after creating /tmp/testdir).

### Expected behavior
Luarocks downloads `busted` and its dependencies and runs the tests.

### Notify maintainers

@teto

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.108, NixOS, 22.11 (Raccoon), 22.11.20230430.db1e4ee`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.11.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

This looks to me like an issue with how luarocks is packaged.
I've been playing around with it for a bit (adding `--set RT_DIR "${pkgs.glibc}" ` to the `wrapProgram` call in the `postInstall` stage), but haven't come up with a working fix yet.

### PR DESCRIPTION
**Pull request by mrcjkb**
###### Description of changes

Fixes #229291.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/luarocks/default.nix [hunk 1]
```diff
@@ -71,6 +72,7 @@ stdenv.mkDerivation (finalAttrs: {
     for i in "$out"/bin/*; do
         test -L "$i" || {
             wrapProgram "$i" \
+              --append-flags "RT_DIR=${glibc}" \
```
Comment (mrcjkb) [inline pkgs/development/tools/misc/luarocks/default.nix:75]: TODO: This will probably not work on darwin.

**Comment by teto** [comment]
This looks ok. Another approach might be to add it to the generated config in `result/etc/luarocks/config-5.2.lua` . I am not sure but that way I think someone might be able to override it.

**Comment by mrcjkb** [comment]
I've opened https://github.com/luarocks/luarocks/issues/1505 and will wait for feedback.

**Comment by mrcjkb** [comment]
Closing this, because I don't think it's needed and there are other packages that can fail to install due to various libraries not being found.
This is not a sustainable solution.

**Pull request closed without merging**